### PR TITLE
send communication method

### PR DESCRIPTION
**Pull request by babasahebk**
Send COMMUNICATION_METHOD as 0 for synchronous processing
Error: ```XML Parsing Failed: COMMUNICATION_METHOD tag in header is missing or value should be ,1,```

### CODE REVIEW

File: lib/conduit/plintron/views/layout.erb [hunk 1]
```diff
@@ -6,6 +6,9 @@
 <% if asynchronous %>
 <COMMUNICATION_METHOD>1</COMMUNICATION_METHOD>
 <% end %>
+<% if send_communication_method %>
```
Comment (dougperez) [inline lib/conduit/plintron/views/layout.erb]: Should this just be a `else`?

Comment (babasahebk) [reply lib/conduit/plintron/views/layout.erb]: No I'm not sure it's impact on other api's so keeping it specific to activate api's

Comment (babasahebk) [reply lib/conduit/plintron/views/layout.erb]: image

So as per this, We really do not need it for other api's.

Comment (pedromartinez) [reply lib/conduit/plintron/views/layout.erb]: I agree this reads a little confusing. How about we define a method called `communication_method` on the actions that need it (which will return integer 0 or 1) and set it to nil in the base class. Then, we can do this:

```
<% if communication_method %>
<COMMUNICATION_METHOD><%= communication_method %></COMMUNICATION_METHOD>
<% end %>
```

File: lib/conduit/plintron/views/layout.erb [hunk 2]
```diff
@@ -6,6 +6,9 @@
 <% if asynchronous %>
 <COMMUNICATION_METHOD>1</COMMUNICATION_METHOD>
```
Comment (dougperez) [inline lib/conduit/plintron/views/layout.erb]: We could use the decorator like we do for other things.


```
    def communication_method
      "1"
    end

    def communication_method
      synchronous? ? "1" : "0"
    end
```

Then use `communication_method` in the layout.

File: spec/conduit/plintron/actions/activate_spec.rb [hunk 3]
```diff
@@ -19,8 +19,17 @@
...
+    context "synchronous processing" do
+      subject { described_class.new(creds.merge(process_synchronously: true)) }
+      it "matches fixture" do
+        xml = File.read("./spec/fixtures/requests/activate/synchronous_request.xml")
+        expect(subject.view).to eql xml
+      end
+    end
   end
 
+
```
Comment (dougperez) [inline spec/conduit/plintron/actions/activate_spec.rb]: Extra space

File: lib/conduit/plintron/views/layout.erb (hunk 1 above)
Comment (dougperez) [reply lib/conduit/plintron/views/layout.erb]: Ok

File: lib/conduit/plintron/views/layout.erb (hunk 2 above)
Comment (dougperez) [reply lib/conduit/plintron/views/layout.erb]: Oh ha, looks like @pedromartinez suggested it also.

File: lib/conduit/plintron/views/layout.erb [hunk 4]
```diff
@@ -5,9 +5,10 @@
 <CHANNEL_REFERENCE><%= channel_reference %></CHANNEL_REFERENCE>
 <% if asynchronous %>
 <COMMUNICATION_METHOD>1</COMMUNICATION_METHOD>
+<% else %>
+<% if communication_method %>
+<COMMUNICATION_METHOD><%= communication_method %></COMMUNICATION_METHOD>
 <% end %>
```
Comment (pedromartinez) [inline lib/conduit/plintron/views/layout.erb]: This looks good, but I bet we can get rid / consolidate line 6 as well.

Comment (babasahebk) [reply lib/conduit/plintron/views/layout.erb]: done

**Pull request merged**